VERSIONING.md

```diff
@@ -0,0 +1,220 @@
+          github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2 v2.0.1`) and in the
+          package import path (e.g., `import
+          "github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2"`). This includes
+          the paths used in `go get` commands (e.g., `go get
+          github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2@v2.0.1`.  Note there
+          is both a `/v2` and a `@v2.0.1` in that example. One way to think about
+          it is that the module name now includes the `/v2`, so include `/v2`
+          whenever you are using the module name).
+        * If a module is version `v0` or `v1`, do not include the major version
+          in either the module path or the import path.
+    * In addition to public APIs, telemetry produced by stable instrumentation
+      will remain stable and backwards compatible. This is to avoid breaking
+      alerts and dashboard.
+    * Modules will be used to encapsulate receivers, processor, exporters,
+      extensions, and any other independent sets of related components.
+        * Experimental modules still under active development will be versioned at
+          `v0` to imply the stability guarantee defined by
+          [semver](https://semver.org/spec/v2.0.0.html#spec-item-4).
+
+          > Major version zero (0.y.z) is for initial development. Anything MAY
+          > change at any time. The public API SHOULD NOT be considered stable.
+
+        * Mature modules for which we guarantee a stable public API and telemetry will
+          be versioned with a major version greater than `v0`.
+        * Experimental modules will start their versioning at `v0.0.0` and will
+          increment their minor version when backwards incompatible changes are
+          released and increment their patch version when backwards compatible
+          changes are released.
+        * Stable contrib modules cannot depend on experimental modules from this
+          project.
+        * All stable contrib modules of the same major version with this project
+          will use the same entire version as this project.
+            * Stable modules may be released with an incremented minor or patch
+              version even though that module's code has not been changed. Instead
+              the only change that will have been included is to have updated that
+              modules dependency on this project's stable APIs.
+            * When an experimental module in contrib becomes stable a new stable
+              module version will be released and will include this now stable
+              module. The new stable module version will be an increment of the minor
+              version number and will be applied to all existing stable contrib
+              modules, this project's modules, and the newly stable module being
+              released.
```

Comment:
Does this mean that when a component moves to v2 it will become "v2.12.3" if the main version of the project is "v1.12.3"? Please add an example to clarify.

Reply:
If there were already other components at `v2.12.3` and a component newly moved to `v2`, then I would expect the next version to be `v2.13.0`.  If it was the first component at `v2` I would expect the version to be `v2.0.0`.  I will add examples.